core/src/main/kotlin/agent/Extensions.kt

```diff
@@ -67,15 +59,16 @@ internal fun AgentInfo.toDto(
     agentManager: AgentManager,
 ): AgentInfoDto = run {
     val plugins = agentManager.plugins.ofAgent(this)
-    val instanceIds = agentManager.instanceIds(id).keys
+    val instanceIds = agentManager.instanceIds(id).keys.toSet()
```

Comment:
why u do keys.toSet ? 
```suggestion
    val instanceIds = agentManager.instanceIds(id).keys
```